test/integration/state-engine.spec.ts

```diff
@@ -134,6 +134,116 @@ describe('state engine', () => {
 		]);
 	});
 
+	it('recovers from a device already running one of the target release services', async () => {
```

Comment:
Without the context of issue 1576, this unit test description is a bit vague. Perhaps could add a reference to the issue in a comment in the code?

Reply:
Added the comment and improved the test description (i think). Let me know what you think

Reply:
@pipex The change looks good!